feat: add `Type::as_string` (#5871)
noir-lang/noir: docs/docs/noir/standard_library/meta/typ.md

If this is a slice type, return the element type of the slice.

> ### as_string
> 
> #include_code as_string noir_stdlib/src/meta/typ.nr rust
> 
> If this is a string type, returns its length as a type.

Reviewer: ```suggestion
### as_str

#include_code as_str noir_stdlib/src/meta/typ.nr rust

If this is a `str<N>` type, returns the length `N` as a type.
```
